Lib/wsgiref/handlers.py

```diff
@@ -179,8 +179,14 @@ def finish_response(self):
                 for data in self.result:
                     self.write(data)
                 self.finish_content()
-        finally:
-            self.close()
+        except:
+            if hasattr(self.result, 'close'):
+                self.result.close()
+            raise
+        # We only call close() when no exception is raised, because it
```

Comment:
Shouldn't this be in an `else` block then?

Reply:
What's the advantage of using an `else` clause in this specific case? Would it make code more readable? Would it make diff shorter? I like using `try...else` clauses, I just want to know why do you prefer the `try...else` way.